build-tools/packages/build-tools/src/fluidBuild/fluidUtils.ts

```diff
@@ -140,27 +144,41 @@ const configExplorer = cosmiconfigSync(configName, {
 	packageProp: [configName],
 });
 
+/**
+ * Set to true when the default config is returned by getFluidBuildConfig so that repeated calls to the function don't
+ * result in repeated searches for config.
+ */
+let defaultConfigLoaded = false;
+
 /**
  * Get an IFluidBuildConfig from the fluidBuild property in a package.json file, or from fluidBuild.config.[c]js.
  *
  * @param rootDir - The path to the root package.json to load.
  * @param noCache - If true, the config cache will be cleared and the config will be reloaded.
- * @returns The fluidBuild section of the package.json, or undefined if not found
+ * @returns The the loaded fluidBuild config, or the default config if one is not found.
  */
 export function getFluidBuildConfig(
 	rootDir: string,
 	noCache = false,
 	log = defaultLogger,
 ): IFluidBuildConfig {
+	if (defaultConfigLoaded) {
+		return DEFAULT_FLUIDBUILD_CONFIG;
+	}
+
```

Comment:
At a glance, we don't use the `nocache` parameter anywhere we call `getFluidBuildConfig` and to me it feels like the new behavior conflicts with the possibility of `noCache = true` (in which case it should have precedence and we should search for config again even if the first time we returned the default config). That said, do we know if this build-tools is actually leveraged by anything outside the repo? I think I remember it being theoretically reusable across projects, but I'd like to question if that's really the case. And if not, maybe get rid of `noCache` to solve the potentially funky interaction?

Reply:
I removed `nocache`.

> That said, do we know if this build-tools is actually leveraged by anything outside the repo?

fluid-build is used in https://github.com/microsoft/TypeAgent/tree/main/ts and
https://github.com/tylerbutler/tools-monorepo. I'd also like to use it in our examples repo, but it's less important
there since there aren't any interdependencies.

From a design standpoint, I generally prefer to design for external usage even if the primary use is internal, because I
think it leads to better code overall.